### PR TITLE
webrender: implement x-create-frame

### PR DESCRIPTION
**Pull request by harryfei**

### CODE REVIEW

File: rust_src/src/wrterm.rs [hunk 1]
```diff
@@ -133,12 +145,78 @@ pub extern "C" fn x_focus_frame(frame: LispFrameRef, noactivate: bool) {
     unimplemented!();
 }
 
+// FRAME is used only to get a handle on the X display.  We don't pass the
+// display info directly because we're called from frame.c, which doesn't
+// know about that structure.
+#[no_mangle]
+pub extern "C" fn x_get_focus_frame(frame: LispFrameRef) -> LispObject {
```
Comment (brotzeit) [inline rust_src/src/wrterm.rs:152]: I think you have to use `*mut frame` here.

Comment (harryfei) [reply rust_src/src/wrterm.rs:152]: We defined those in rust code.
```rust
#[repr(transparent)]
pub struct ExternalPtr<T>(*mut T);

pub type LispFrameRef = ExternalPtr<Lisp_Frame>; 

```

So `LispFrameRef` just equals `*mut frame`.

Comment (brotzeit) [reply rust_src/src/wrterm.rs:152]: I noticed that we used the C types in remacs so I thought we did it for a reason.

Comment (harryfei) [reply rust_src/src/wrterm.rs:152]: I think it's safe. Some functions in remacs use this trick too.

https://github.com/remacs/remacs/blob/master/rust_src/src/dispnew.rs#L65
```rust
/// Redraw frame FRAME.
#[no_mangle]
pub extern "C" fn redraw_frame(mut frame: LispFrameRef) {
```

**Comment by harryfei** [comment]
Merge? @brotzeit

**Review by brotzeit — APPROVED** [review]

**Pull request merged**